block-layer/codecs/dag-json.md

```diff
@@ -16,24 +24,25 @@ Codec implementors **MUST** do the following in order to ensure hashes consisten
 This produces the most compact and consistent representation which will ensure that two codecs
 producing the same data end up with matching block hashes.
 
-### Natively supported kinds
+## Supported Kinds
+
+All [IPLD Data Model Kinds](../../data-model.md#kinds) except Bytes and Link are supported natively by JSON.
 
-All kinds of the IPLD Data Model except Bytes and Link are supported natively by JSON.
+Bytes and Links use extensions specific to DAG-JSON. They are implemented as an map, where the single key is a slash (`"/"`) and the value contains the kind's data.
 
-#### Numbers
+### Numbers
 
-Numbers are a special case. JSON only has a single number type, though many dynamically typed programming languages (e.g. Python, Ruby, PHP) distinguish between integers and floats when parsing JSON. A number consisting of an optional leading sign and only digits is parsed as integer, if it contains a decimal point, it's parsed as a float. In DAG-JSON the same method is used to represent integers and floats.
+JSON only has a single number type. Many dynamically typed programming languages (e.g. Python, Ruby, PHP) distinguish between integers and floats when parsing JSON. JavaScript does not since all numbers are represented internally as IEEE 754 floats. A JSON number consisting of an optional leading sign (`-`) and only digits is parsed as integer, if it contains a decimal point, it's parsed as a float. DAG-JSON the same method is used to represent integers and floats.
 
-Contrary to popular belief, JSON as a format supports Big Integers. It's only
-JavaScript itself that has trouble with them. This means JS implementations
-of DAG-JSON can't use the native JSON parser and serializer if integers bigger
-than `2^53 - 1` should be supported.
+Data Model floats that do not have a fractional component should be encoded **with** a decimal point, and will therefore be distinguishable from an integer during round-trip. (Note that since JavaScript still cannot distinguish a float from an integer where the number has no fractional component, this rule will not impact JavaScript encoding or decoding).
```

Comment:
I'm not sure if the note about JavaScript is needed as it does exactly that when encoding/decoding JSON:

```console
> JSON.stringify(2.1)
'2.1'
> JSON.stringify(2.0)
'2'
> JSON.parse('2.1')
2.1
> JSON.parse('2.0')
2
```

Reply:
yeah, you're making my point @vmx. The spec is saying that `2.0` should be encoded as `2.0` but JavaScript can only do `2` since we can't distinguish. Then in the reverse the spec says that a `2.0` should be decoded as a "float" but JavaScript still can't distinguish so a `2` and a `2.0` would come out as the same thing, regardless. Maybe one day we'll retain the information that "this was a float in the encoded form" but there's no infrastructure to attach that information today.

So the note basically says "JavaScript can't conform to this part of the spec". The spec says "... floats that **do not** have a fractional component should be encoded **with** a decimal point".